src/test/acceptance/partner_profile.js

```diff
@@ -3,37 +3,66 @@ import { setup, teardown, stubAuctionReminder } from './helpers'
 import { server as antigravity } from 'antigravity'
 
 describe('Partner profile page', () => {
-  let gravity, browser
+  let gravity, positron, browser
 
   before(async () => {
-    ;({ gravity, browser } = await setup())
+    ;({ gravity, positron, browser } = await setup())
     stubAuctionReminder()
+    gravity.get('/api/v1/shortcut/:id', (req, res) => {
+      res.res.sendStatus(404)
+    })
     gravity.use(antigravity)
-    await browser.useragent('iPhone')
   })
 
   after(teardown)
 
-  it('shows the list of shows', async () => {
-    const $ = await browser.page('/gagosian-gallery/shows')
-    $.html().should.containEql('Upcoming Shows')
+  context('desktop', () => {
+    it('shows the overview', async () => {
+      const $ = await browser.page('/gagosian-gallery')
+      $.html().should.containEql(
+        '<h1 class="partner-name">Gagosian Gallery</h1>'
+      )
+      $.html().should.containEql('<p class="partner-bio">Living the dream as')
+    })
   })
 
-  it('shows partner articles', async () => {
-    const $ = await browser.page('/gagosian-gallery/articles')
-    $.html().should.containEql('Articles')
-  })
+  context('iPhone', () => {
+    before(() => {
+      browser.useragent('iPhone')
+      positron.get('/api/articles', (req, res) => {
+        res.send(require('./fixtures/gravity/home'))
+      })
+    })
 
-  it('does not show contact information for non-active partner', async () => {
-    const $ = await browser.page('/gagosian-gallery/contact')
-    $.html().should.containEql(
-      'Sorry, the page you were looking for doesn&#x2019;t exist at this URL.'
-    )
-  })
+    it('shows the overview', async () => {
+      const $ = await browser.page('/gagosian-gallery?_')
```

Comment:
Looks like Nightmare would cache pages and the result of a previous test above would leak here and fail this spec. Moving all the setup and teardown into `beforeEach` from `before` (in each test we get a clean browser, etc) would solve this, but I feel it's a waste of resources. I decided to simply append a random param to avoid the caching.

I did try the `clearCache` approach given in the [examples](https://github.com/segmentio/nightmare#extending-nightmare) but couldn't make it work. 🤷‍♂️ Will save this for a later time.